File: ioi/2013/day0-1/citizen/grader.c
#include <stdio.h>
#include <stdlib.h>
#include "citizen.h"

#define fail(s, x...) do { \
		fprintf(stderr, s "\n", ## x); \
		exit(1); \
	} while(0)

#define MAX_N 100000

int P[MAX_N];
int Q[MAX_N];

int main() {
	int N, i;
	int res;

	FILE *f = fopen("citizen.in", "r");
	if (!f)
		fail("Failed to open input file.");

	res = fscanf(f, "%d", &N);
	if (res != 1)
		fail("Failed to read N from input file.");

	if (N < 1 || N > MAX_N)
		fail("N is out of bounds.");

	for (i = 0; i < N; i++) {
		res = fscanf(f, "%d%d", &P[i], &Q[i]);
		if (res != 2)
			fail("Failed to read data from input file.");
	}
	fclose(f);

	int answer = countries(N, P, Q);

	printf("%d\n", answer);

	return 0;
}

File: ioi/2013/day0-1/citizen/grader.pas
program grader;

uses citizen;

procedure fail(s: string);
begin
    writeln(stderr, s);
    halt(1);
end;

var
    N, i: longint;
    P, Q: array of Longint;
    f: text;
    answer: longint;

begin
    assign(f, 'citizen.in');
    {$I-} reset(f); {$I+}
    if IOResult <> 0 then
        fail('Failed to open input file.');

    readln(f, N);
    setlength(P, N);
    setlength(Q, N);

    for i := 0 to N-1 do
        readln(f, P[i], Q[i]);
    close(f);

    answer := countries(N, P, Q);

    writeln(answer);
end.
